src/vm/eventpipe.cpp

```diff
@@ -582,30 +585,21 @@ void EventPipe::WriteEventInternal(
         BYTE *pData = payload.GetFlatData();
         if (pThread != nullptr && pRundownSession != nullptr && pData != nullptr)
         {
-            // Write synchronously to the file.
-            // We're under lock and blocking the disabling thread.
-            // This copy occurs here (rather than at file write) because
-            // A) The FastSerializer API would need to change if we waited
-            // B) It is unclear there is a benefit to multiple file write calls
-            //    as opposed a a buffer copy here
-            EventPipeEventInstance instance(
-                event,
-                GetCurrentProcessorNumber(),
-                pEventPipeThread->GetOSThreadId(),
-                pData,
-                payload.GetSize(),
-                pActivityId,
-                pRelatedActivityId);
-            instance.EnsureStack(*pRundownSession);
-
             // EventPipeFile::WriteEvent needs to allocate a metadata event
             // and can therefore throw. In this context we will silently
             // fail rather than disrupt the caller
             EX_TRY
             {
                 _ASSERTE(pRundownSession != nullptr);
                 if (pRundownSession != nullptr)
-                    pRundownSession->WriteEventUnbuffered(instance, pEventPipeThread);
```

Comment:
I think you just made WriteEventUnbuffered into dead code in which case lets delete it : )